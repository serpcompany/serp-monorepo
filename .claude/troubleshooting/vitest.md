https://vitest.dev/guide/common-errors.html