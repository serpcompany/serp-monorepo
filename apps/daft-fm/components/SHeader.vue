<!-- components/SHeader.vue -->
<template>
  <div class="container mx-auto my-4 px-4 py-4">
    <div class="flex items-center justify-between">
      <!-- left: brand logo -->
      <div>
        <nuxt-link to="/" aria-label="homepage">
          <s-logo />
        </nuxt-link>
      </div>

      <!-- middle: navigation -->
      <!-- nav links -->
      <div>
        <u-navigation-menu :items="items" />
      </div>

      <!-- right: profile avatar thing -->
      <div>
        <!-- <u-avatar /> -->
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
const items = [
  {
    label: 'Artists',
    to: '/artists'
  },
  {
    label: 'Albums',
    to: '/albums'
  },
  {
    label: 'Songs',
    to: '/songs'
  }
]
</script>
