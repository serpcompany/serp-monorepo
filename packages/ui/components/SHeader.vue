<template>
  <header>
    <div>
      <div class="flex h-16 items-center justify-between">
        <!-- Logo on left -->
        <div class="flex items-center">
          <div class="flex-shrink-0">
            <nuxt-link to="/" aria-label="home"><s-logo /></nuxt-link>
          </div>

          <!-- Navigation for larger screens -->
          <div class="ml-6 hidden lg:flex">
            <UNavigationMenu
              aria-label="Main navigation"
              orientation="horizontal"
              highlight
              :items="headerNavItems"
              :ui="{
                content: 'absolute top-0',
                list: 'flex',
                childList: 'grid grid-cols-1',
                viewportWrapper:
                  'absolute top-full translate-x-[var(--radix-navigation-menu-viewport-position-x)]',
                viewport:
                  'relative overflow-hidden bg-(--ui-bg) shadow-lg rounded-[calc(var(--ui-radius)*1.5)] ring ring-(--ui-border) h-(--reka-navigation-menu-viewport-height) min-w-[180px] transition-[height] duration-200 origin-[top_left] data-[state=open]:animate-[scale-in_100ms_ease-out] data-[state=closed]:animate-[scale-out_100ms_ease-in]'
              }"
              :content="{
                align: 'center',
                side: 'bottom',
                sideOffset: 8
              }"
            />
          </div>
        </div>

        <!-- Right side items -->
        <div class="flex items-center">
          <!-- Hamburger menu for small screens -->
          <div class="lg:hidden">
            <button
              type="button"
              class="rounded-md p-2 hover:text-[var(--ui-primary)] focus:ring-2 focus:ring-[var(--ui-primary)] focus:outline-none focus:ring-inset"
              :aria-expanded="mobileMenuOpen"
              @click="toggleMobileMenu"
            >
              <span class="sr-only">Open main menu</span>
              <svg
                class="h-6 w-6"
                xmlns="http://www.w3.org/2000/svg"
                fill="none"
                viewBox="0 0 24 24"
                stroke="currentColor"
                aria-hidden="true"
              >
                <path
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  d="M4 6h16M4 12h16M4 18h16"
                />
              </svg>
            </button>
          </div>

          <!-- Color mode button on right for larger screens -->
          <div class="hidden items-center space-x-4 lg:flex">
            <color-mode-button />
            <profile-dropdown v-if="useAuth" />
          </div>
        </div>
      </div>
    </div>

    <!-- Mobile menu -->
    <div v-if="mobileMenuOpen" class="lg:hidden">
      <div class="space-y-1 px-2 pt-2 pb-3">
        <UNavigationMenu
          orientation="vertical"
          highlight
          :items="headerNavItems"
          class="w-full"
          :ui="{
            content: 'absolute top-0',
            childList: 'grid grid-cols-1',
            viewportWrapper:
              'absolute top-full translate-x-[var(--radix-navigation-menu-viewport-position-x)]',
            viewport:
              'relative overflow-hidden bg-(--ui-bg) shadow-lg rounded-[calc(var(--ui-radius)*1.5)] ring ring-(--ui-border) h-(--reka-navigation-menu-viewport-height) min-w-[180px] transition-[height] duration-200 origin-[top_left] data-[state=open]:animate-[scale-in_100ms_ease-out] data-[state=closed]:animate-[scale-out_100ms_ease-in]'
          }"
          :content="{
            align: 'center',
            side: 'bottom',
            sideOffset: 8
          }"
        />
        <div class="mt-4 border-t border-gray-200 pt-4">
          <profile-dropdown />
        </div>
      </div>
    </div>
  </header>
</template>

<script setup>
const config = useRuntimeConfig();
const headerNavItems = config.public.headerNavItems;
const useAuth = config.public.useAuth;

const mobileMenuOpen = ref(false);

const toggleMobileMenu = () => {
  mobileMenuOpen.value = !mobileMenuOpen.value;
};
</script>
