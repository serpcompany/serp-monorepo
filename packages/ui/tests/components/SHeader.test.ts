import { mockNuxtImport } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'
import SHeader from '../../components/SHeader.vue'
import ComponentRender from '../componentRender'
import '../mockUseUserSession'

// We're testing different runtime config setups to ensure SHeader renders correctly under various conditions.
let config_: Record<string, unknown> = {
    public: {
        headerNavItems: [
        { name: 'Home', href: '/' },
        { name: 'About', href: '/about' }
        ],
        useAuth: true
    }
}

describe('SHeader Snapshot', () => {
  // Test scenarios with different runtime config values.
  const scenarios: [string, { config: Record<string, unknown> }][] = [
    [
      'with auth enabled',
      {
        config: {
          public: {
            headerNavItems: [
              { name: 'Home', href: '/' },
              { name: 'About', href: '/about' }
            ],
            useAuth: true
          }
        }
      }
    ],
    [
      'with auth disabled',
      {
        config: {
          public: {
            headerNavItems: [
              { name: 'Home', href: '/' },
              { name: 'About', href: '/about' }
            ],
            useAuth: false
          }
        }
      }
    ],
    [
      'with empty navigation items',
      {
        config: {
          public: {
            headerNavItems: [],
            useAuth: true
          }
        }
      }
    ]
  ]

  it.each(scenarios)('renders %s correctly', async (desc: string, { config }) => {
    // Set up the runtime config for this test scenario
    config_ = config
    mockNuxtImport('useRuntimeConfig', () => () => config_)

    const html = await ComponentRender(`SHeader ${desc}`, {}, SHeader)
    expect(html).toMatchSnapshot()
  })
})
