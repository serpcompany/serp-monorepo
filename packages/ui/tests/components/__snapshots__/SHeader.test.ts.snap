// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SHeader Snapshot > renders with auth disabled correctly 1`] = `
"<header class="bg-background top-0 z-50 mx-auto h-18 max-w-full border-b border-(--ui-border) px-12">
  <div class="flex h-full items-center justify-between px-4">
    <!-- Left side -->
    <div class="flex items-center">
      <!-- Logo -->
      <div class="flex-shrink-0"><a href="/" aria-label="home">
          <div class="flex items-center space-x-2"><svg width="35pt" viewBox="0 0 1024 1024" xmlns="http://www.w3.org/2000/svg" aria-label="SERP Brand logo">
              <path class="fill-[var(--ui-primary)]" d=" M 127.62 540.68 C 255.64 429.97 383.70 319.31 511.76 208.65 C 639.74 319.23 767.70 429.86 895.69 540.45 C 895.70 631.71 895.73 722.97 895.64 814.23 C 719.72 662.17 543.76 510.14 367.81 358.10 C 398.86 414.80 429.83 471.54 460.92 528.22 C 349.85 623.79 238.69 719.27 127.67 814.91 C 127.66 723.50 127.67 632.09 127.62 540.68 Z"></path>
            </svg></div>
        </a></div><!-- Desktop nav -->
      <div class="ml-6 hidden lg:flex">
        <!---->
        <!---->
        <nav aria-label="Main" data-orientation="horizontal" dir="ltr" data-reka-navigation-menu="" data-collapsed="false" class="relative flex gap-1.5 [&amp;>div]:min-w-0 items-center justify-between">
          <div style="position: relative;">
            <ul class="isolate min-w-0 flex items-center" data-orientation="horizontal">
              <li data-menu-item="" class="min-w-0 py-2"><a href="/" class="group focus-visible:before:ring-(--ui-primary) px-2.5 py-1.5 before:inset-x-px before:inset-y-0 after:absolute after:-bottom-2 after:inset-x-2.5 after:block after:h-px after:rounded-full after:transition-colors hover:text-(--ui-text-highlighted) hover:before:bg-(--ui-bg-elevated)/50 transition-colors before:transition-colors data-[state=open]:text-(--ui-text-highlighted) data-[state=open]:before:bg-(--ui-bg-elevated)/50 group relative w-full flex items-center gap-1.5 font-medium text-sm before:absolute before:z-[-1] before:rounded-md focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:before:ring-inset focus-visible:before:ring-2 text-primary" data-reka-collection-item="">
                  <!--v-if-->
                  <!--v-if-->
                  <!--v-if-->
                </a>
                <!--v-if-->
              </li>
              <li data-menu-item="" class="min-w-0 py-2"><a href="/about" class="group focus-visible:before:ring-(--ui-primary) px-2.5 py-1.5 before:inset-x-px before:inset-y-0 after:absolute after:-bottom-2 after:inset-x-2.5 after:block after:h-px after:rounded-full after:transition-colors hover:text-(--ui-text-highlighted) hover:before:bg-(--ui-bg-elevated)/50 transition-colors before:transition-colors data-[state=open]:text-(--ui-text-highlighted) data-[state=open]:before:bg-(--ui-bg-elevated)/50 group relative w-full flex items-center gap-1.5 font-medium text-sm before:absolute before:z-[-1] before:rounded-md focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:before:ring-inset focus-visible:before:ring-2 text-primary" data-reka-collection-item="">
                  <!--v-if-->
                  <!--v-if-->
                  <!--v-if-->
                </a>
                <!--v-if-->
              </li>
            </ul>
          </div>
          <!--v-if-->
          <div class="absolute top-full left-0 flex w-full">
            <!--v-if-->
            <!---->
          </div>
        </nav>
      </div>
    </div><!-- Right side -->
    <div class="flex items-center">
      <!-- Hamburger menu for small screens -->
      <div class="lg:hidden"><button type="button" class="rounded-md p-2 hover:text-[var(--ui-primary)] focus:ring-2 focus:ring-[var(--ui-primary)] focus:outline-none focus:ring-inset" aria-expanded="false"><span class="sr-only">Open main menu</span><svg class="h-6 w-6" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true">
            <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M4 6h16M4 12h16M4 18h16"></path>
          </svg></button></div><!-- Desktop color mode & profile -->
      <div class="hidden items-center space-x-4 lg:flex"><button type="button" aria-label="Toggle color mode" class="rounded-[calc(var(--ui-radius)*1.5)] font-medium inline-flex items-center focus:outline-hidden disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors text-sm gap-1.5 text-(--ui-text) hover:bg-(--ui-bg-elevated) focus-visible:bg-(--ui-bg-elevated) hover:disabled:bg-transparent dark:hover:disabled:bg-transparent hover:aria-disabled:bg-transparent dark:hover:aria-disabled:bg-transparent p-1.5"><span class="iconify i-lucide:sun shrink-0 size-5" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button>
        <!--v-if-->
      </div>
    </div>
  </div><!-- Mobile menu -->
  <!--v-if-->
</header>
<div class="px-12">
  <!--v-if-->
</div>"
`;

exports[`SHeader Snapshot > renders with auth enabled correctly 1`] = `
"<header class="bg-background top-0 z-50 mx-auto h-18 max-w-full border-b border-(--ui-border) px-12">
  <div class="flex h-full items-center justify-between px-4">
    <!-- Left side -->
    <div class="flex items-center">
      <!-- Logo -->
      <div class="flex-shrink-0"><a href="/" aria-label="home">
          <div class="flex items-center space-x-2"><svg width="35pt" viewBox="0 0 1024 1024" xmlns="http://www.w3.org/2000/svg" aria-label="SERP Brand logo">
              <path class="fill-[var(--ui-primary)]" d=" M 127.62 540.68 C 255.64 429.97 383.70 319.31 511.76 208.65 C 639.74 319.23 767.70 429.86 895.69 540.45 C 895.70 631.71 895.73 722.97 895.64 814.23 C 719.72 662.17 543.76 510.14 367.81 358.10 C 398.86 414.80 429.83 471.54 460.92 528.22 C 349.85 623.79 238.69 719.27 127.67 814.91 C 127.66 723.50 127.67 632.09 127.62 540.68 Z"></path>
            </svg></div>
        </a></div><!-- Desktop nav -->
      <div class="ml-6 hidden lg:flex">
        <!---->
        <!---->
        <nav aria-label="Main" data-orientation="horizontal" dir="ltr" data-reka-navigation-menu="" data-collapsed="false" class="relative flex gap-1.5 [&amp;>div]:min-w-0 items-center justify-between">
          <div style="position: relative;">
            <ul class="isolate min-w-0 flex items-center" data-orientation="horizontal">
              <li data-menu-item="" class="min-w-0 py-2"><a href="/" class="group focus-visible:before:ring-(--ui-primary) px-2.5 py-1.5 before:inset-x-px before:inset-y-0 after:absolute after:-bottom-2 after:inset-x-2.5 after:block after:h-px after:rounded-full after:transition-colors hover:text-(--ui-text-highlighted) hover:before:bg-(--ui-bg-elevated)/50 transition-colors before:transition-colors data-[state=open]:text-(--ui-text-highlighted) data-[state=open]:before:bg-(--ui-bg-elevated)/50 group relative w-full flex items-center gap-1.5 font-medium text-sm before:absolute before:z-[-1] before:rounded-md focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:before:ring-inset focus-visible:before:ring-2 text-primary" data-reka-collection-item="">
                  <!--v-if-->
                  <!--v-if-->
                  <!--v-if-->
                </a>
                <!--v-if-->
              </li>
              <li data-menu-item="" class="min-w-0 py-2"><a href="/about" class="group focus-visible:before:ring-(--ui-primary) px-2.5 py-1.5 before:inset-x-px before:inset-y-0 after:absolute after:-bottom-2 after:inset-x-2.5 after:block after:h-px after:rounded-full after:transition-colors hover:text-(--ui-text-highlighted) hover:before:bg-(--ui-bg-elevated)/50 transition-colors before:transition-colors data-[state=open]:text-(--ui-text-highlighted) data-[state=open]:before:bg-(--ui-bg-elevated)/50 group relative w-full flex items-center gap-1.5 font-medium text-sm before:absolute before:z-[-1] before:rounded-md focus:outline-none focus-visible:outline-none dark:focus-visible:outline-none focus-visible:before:ring-inset focus-visible:before:ring-2 text-primary" data-reka-collection-item="">
                  <!--v-if-->
                  <!--v-if-->
                  <!--v-if-->
                </a>
                <!--v-if-->
              </li>
            </ul>
          </div>
          <!--v-if-->
          <div class="absolute top-full left-0 flex w-full">
            <!--v-if-->
            <!---->
          </div>
        </nav>
      </div>
    </div><!-- Right side -->
    <div class="flex items-center">
      <!-- Hamburger menu for small screens -->
      <div class="lg:hidden"><button type="button" class="rounded-md p-2 hover:text-[var(--ui-primary)] focus:ring-2 focus:ring-[var(--ui-primary)] focus:outline-none focus:ring-inset" aria-expanded="false"><span class="sr-only">Open main menu</span><svg class="h-6 w-6" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true">
            <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M4 6h16M4 12h16M4 18h16"></path>
          </svg></button></div><!-- Desktop color mode & profile -->
      <div class="hidden items-center space-x-4 lg:flex"><button type="button" aria-label="Toggle color mode" class="rounded-[calc(var(--ui-radius)*1.5)] font-medium inline-flex items-center focus:outline-hidden disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors text-sm gap-1.5 text-(--ui-text) hover:bg-(--ui-bg-elevated) focus-visible:bg-(--ui-bg-elevated) hover:disabled:bg-transparent dark:hover:disabled:bg-transparent hover:aria-disabled:bg-transparent dark:hover:aria-disabled:bg-transparent p-1.5"><span class="iconify i-lucide:sun shrink-0 size-5" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button>
        <div class="relative">
          <!-- Avatar trigger -->
          <div><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden rounded-full align-middle bg-(--ui-bg-elevated) size-8 text-base cursor-pointer transition-opacity hover:opacity-80"><span class="font-medium leading-none text-(--ui-text-muted) truncate">&nbsp;</span></span></div><!-- Slideover menu -->
          <!--v-if-->
          <!--teleport start-->
          <!--teleport end-->
        </div>
      </div>
    </div>
  </div><!-- Mobile menu -->
  <!--v-if-->
</header>
<div class="px-12">
  <!--v-if-->
</div>"
`;

exports[`SHeader Snapshot > renders with empty navigation items correctly 1`] = `
"<header class="bg-background top-0 z-50 mx-auto h-18 max-w-full border-b border-(--ui-border) px-12">
  <div class="flex h-full items-center justify-between px-4">
    <!-- Left side -->
    <div class="flex items-center">
      <!-- Logo -->
      <div class="flex-shrink-0"><a href="/" aria-label="home">
          <div class="flex items-center space-x-2"><svg width="35pt" viewBox="0 0 1024 1024" xmlns="http://www.w3.org/2000/svg" aria-label="SERP Brand logo">
              <path class="fill-[var(--ui-primary)]" d=" M 127.62 540.68 C 255.64 429.97 383.70 319.31 511.76 208.65 C 639.74 319.23 767.70 429.86 895.69 540.45 C 895.70 631.71 895.73 722.97 895.64 814.23 C 719.72 662.17 543.76 510.14 367.81 358.10 C 398.86 414.80 429.83 471.54 460.92 528.22 C 349.85 623.79 238.69 719.27 127.67 814.91 C 127.66 723.50 127.67 632.09 127.62 540.68 Z"></path>
            </svg></div>
        </a></div><!-- Desktop nav -->
      <div class="ml-6 hidden lg:flex">
        <!---->
        <!---->
        <nav aria-label="Main" data-orientation="horizontal" dir="ltr" data-reka-navigation-menu="" data-collapsed="false" class="relative flex gap-1.5 [&amp;>div]:min-w-0 items-center justify-between">
          <div class="absolute top-full left-0 flex w-full">
            <!--v-if-->
            <!---->
          </div>
        </nav>
      </div>
    </div><!-- Right side -->
    <div class="flex items-center">
      <!-- Hamburger menu for small screens -->
      <div class="lg:hidden"><button type="button" class="rounded-md p-2 hover:text-[var(--ui-primary)] focus:ring-2 focus:ring-[var(--ui-primary)] focus:outline-none focus:ring-inset" aria-expanded="false"><span class="sr-only">Open main menu</span><svg class="h-6 w-6" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true">
            <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M4 6h16M4 12h16M4 18h16"></path>
          </svg></button></div><!-- Desktop color mode & profile -->
      <div class="hidden items-center space-x-4 lg:flex"><button type="button" aria-label="Toggle color mode" class="rounded-[calc(var(--ui-radius)*1.5)] font-medium inline-flex items-center focus:outline-hidden disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors text-sm gap-1.5 text-(--ui-text) hover:bg-(--ui-bg-elevated) focus-visible:bg-(--ui-bg-elevated) hover:disabled:bg-transparent dark:hover:disabled:bg-transparent hover:aria-disabled:bg-transparent dark:hover:aria-disabled:bg-transparent p-1.5"><span class="iconify i-lucide:sun shrink-0 size-5" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button>
        <div class="relative">
          <!-- Avatar trigger -->
          <div><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden rounded-full align-middle bg-(--ui-bg-elevated) size-8 text-base cursor-pointer transition-opacity hover:opacity-80"><span class="font-medium leading-none text-(--ui-text-muted) truncate">&nbsp;</span></span></div><!-- Slideover menu -->
          <!--v-if-->
          <!--teleport start-->
          <!--teleport end-->
        </div>
      </div>
    </div>
  </div><!-- Mobile menu -->
  <!--v-if-->
</header>
<div class="px-12">
  <!--v-if-->
</div>"
`;
